spec/f_spec.rb

```diff
@@ -3,68 +3,68 @@
 include ExampleWithGSL
 
 describe Distribution::F do
-shared_examples_for "F engine(with rng)" do
-  it "should return correct rng" do
-  pending()
+  shared_examples_for "F engine(with rng)" do
+    it "should return correct rng" do
+      pending()
+    end
   end
-end
 
-shared_examples_for "F engine(with pdf)" do
-  it_only_with_gsl "should return correct pdf" do
-    if @engine.respond_to? :pdf
-      [0.1,0.5,1,2,10,20,30].each{|f|
-      [2,5,10].each{|n2|
-      [2,5,10].each{|n1|
-      @engine.pdf(f,n1,n2).should be_within(1e-4).of(GSL::Ran.fdist_pdf(f,n1,n2))
-      }
-      }
-      }
-    else
-      pending("No #{@engine}.pdf")
+  shared_examples_for "F engine(with pdf)" do
+    it_only_with_gsl "should return correct pdf" do
+      if @engine.respond_to? :pdf
+        [0.1,0.5,1,2,10,20,30].each do |f|
+          [2,5,10].each do |n2|
+            [2,5,10].each do |n1|
+              @engine.pdf(f,n1,n2).should be_within(1e-4).of(
+                GSL::Ran.fdist_pdf(f,n1,n2)
+              )
+            end
+          end
+        end
+      else
+        pending("No #{@engine}.pdf")
+      end
     end
   end
-end
 
-shared_examples_for "F engine" do
-  
-  it_only_with_gsl "should return correct cdf" do
-    if @engine.respond_to? :cdf
-      [0.1,0.5,1,2,10,20,30].each{|f|
-      [2,5,10].each{|n2|
-      [2,5,10].each{|n1|
-      @engine.cdf(f,n1,n2).should be_within(1e-4).of(GSL::Cdf.fdist_P(f,n1,n2))
-      
-      }
-      }
-      }
-    else
-      pending("No #{@engine}.cdf")
+  shared_examples_for "F engine" do
+    it_only_with_gsl "should return correct cdf" do
+      if @engine.respond_to? :cdf
+        [0.1,0.5,1,2,10,20,30].each do |f|
+          [2,5,10].each do |n2|
+            [2,5,10].each do |n1|
+              @engine.cdf(f,n1,n2).should be_within(1e-4).of(GSL::Cdf.fdist_P(f,n1,n2))        
+            end
+          end
+        end
+      else
+        pending("No #{@engine}.cdf")
+      end
     end
-  
-  end
-  it_only_with_gsl "should return correct p_value" do
-    if @engine.respond_to? :p_value
-  
-    [0.1,0.5,1,2,10,20,30].each{|f|
-    [2,5,10].each{|n2|
-    [2,5,10].each{|n1|
-    area=@engine.cdf(f,n1,n2)
-    @engine.p_value(area,n1,n2).should be_within(1e-4).of(GSL::Cdf.fdist_Pinv(area,n1,n2))
-    }
-    }
-    }
-  
-    
-    else
-      pending("No #{@engine}.p_value")
+
+    it_only_with_gsl "should return correct p_value", :focus => true do
+      if @engine.respond_to? :p_value
+        [0.1,0.5,1,2,10,20,30].each do |f|
+          [2,5,10].each do|n2|
+            [2,5,10].each do |n1|
+              area = @engine.cdf(f,n1,n2)
+              @engine.p_value(area,n1,n2).should be_within(1e-4).of(GSL::Cdf.fdist_Pinv(area,n1,n2))
+              @engine.p_value(0.975, 5, 4.189092917592713).should be_within(
```

Comment:
@agarie the values that you see here (the arguments to the `p_value` function) are the exact values that [statsample was passing into the function](https://github.com/v0dro/statsample/blob/master/lib/statsample/reliability/icc.rb#L344), which ultimately made it fail.

When I substitute `Distribution::F.p_value` with `GSL::Cdf.fdist_Pinv` in statsample, the test passes. So this is definitely a problem with the ruby implementation of `Distribution::F.p_value`/